tests/job_tests/returnn/test_search.py

```diff
@@ -43,3 +44,16 @@ def test_search_bpe_to_words_nbest():
             open(bpe_to_words_job.out_word_search_results.get_path(), "rt").read()
         )
         assert reference_dict == job_dict
+
+
+def test_corpus_replace_orth_from_reference():
```

Comment:
I think these tests are organized per module. So the test for `i6_core.corpus.convert.CorpusReplaceOrthFromReferenceCorpus` should live in `tests/job_tests/corpus/test_convert.py`